72_config_and_simplify_tls.md

```diff
@@ -0,0 +1,367 @@
+* Provide a migration tool to port common Keylime configurations to the new
+  format
+* Simplify the TLS setup of Keylime
+* Cleanup keylime_ca to make it work with the new setup
+* Formally allow the Rust agent configuration to diverge from the Python agent
+
+### Non-Goals
+
+* Implement other mechanisms of authentication other than mTLS
+* Move registration to HTTPS (but can be easily done in a followup change)
+* Add a migration path for all possible old Keylime configurations
+
+## Proposal - Split the global configuration into the different components and Cleanup
+
+Distributions should ship the default configuration in `/usr/etc/keylime/` or if
+not supported in `/etc/keylime/`. If `/etc/keylime/` is populated the options in
+`/usr/etc/keylime` will be ignored. Manual overrides can be provided in
+`/etc/keylime.d/`.
+
+For the new configuration is split into the following files with the following
+options:
+
+* `global.conf`: Contains global configuration.
+  * `enable_tls`: Enable or disable TLS for the servers (registrar, verifier) and
+    for the tenant if it tries to use TLS. (This setting is currently broken, so
+    we either fix it or fully remove it)
+  * Logging configuration
+  * `version`: Version number of the entire configuration.
+* `agent.conf`: Contains the agent specific configuration.
+  * `ip`:  IP where the agent listens on. (Should this be a list?)
+  * `port`: Port where the agent listens on
+  * `contact_ip` (optional): On which IP can the verifier/tenant contact the agent 
+  * `contact_port` (optional): On which port can the verifier/tenant contact the agent 
+  * `registrar_ip`: IP of the registrar
+  * `registrar_port`: Port of the registrar
+  * `enable_agent_mtls`:  See TLS simplification
+  * `tls_dir`: See TLS simplification
+  * `server_key`: See TLS simplification
+  * `server_cert`: See TLS simplification
+  * `trusted_client_ca`: See TLS simplification
+  * `trusted_server_ca`: See TLS simplification
+  * `enc_keyname`: Derived key K from U/V split for payload
+  * `dec_payload_file`: Name of the decrypted payload
+  * `secure_size`: Size of the tmpfs. Note hat in most cases this is provided by
+    the distribution already.
+  * `tpm_ownerpassword`: (Only Python agent)  TPM owner password (We do not
+    really need taking ownership for remote attestation). Will be probably
+    removed in the future.
+  * `extract_payload_zip`: Try to extract the payload
+  * `uuid`: UUID of the agent or one of the following options
+    * `generate`: creates random UUID
+    * `hash_ek`: Hash of the EK_pub in PEM format
+    * `hostname`: The hostname of the system
+    * `environment`: Uses the environment variable "KEYLIME_AGENT_UUID" as UUID.
+    * `dmidecode`: system UUID (Can we deprecate this option?)
+  * `enable_revocation_notifications`: Option to enable or disable listening for
+    revocation notifications from the verifier.
+  * `revocation_notifier_ip`: IP of the revocation notifier service
```

Comment:
Given that we are considering the possibility of adding more revocation methods (as explicit in the `verifier.conf`, with at least two defined methods, "0MQ" and "webhook"), shouldn't we mirror it in `agent.conf`? I agree that a method such as "webhook" has no direct effect on the agent, but we should at least be open to the possibility of additional revocation methods (some of which might include agent/verifier) later.

Reply:
Yes, with the revocation hooks via REST proposed in https://github.com/keylime/enhancements/blob/master/53_agent_local_revocation.md we should make this more generic for the agent.